src/ts/components/controlbar.ts

```diff
@@ -32,10 +33,42 @@ export class ControlBar extends Container<ControlBarConfig> {
   configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
     super.configure(player, uimanager);
 
+    // Counts how many components are hovered and block hiding of the control bar
+    let hoverStackCount = 0;
+
+    if (uimanager.getConfig().disableAutoHideWhenHovered) {
+      // only enabling this for non-mobile platforms without touch input. enabling this
+      // for touch devices causes the UI to not disappear after the standard few seconds.
+      // Instead, it will stay visible until another manual action is performed.
+      if (!BrowserUtils.isMobile) {
```

Comment:
These statements can be combined:
```ts
    // only enabling this for non-mobile platforms without touch input. enabling this
    // for touch devices causes the UI to not disappear after the standard few seconds.
    // Instead, it will stay visible until another manual action is performed.
    if (uimanager.getConfig().disableAutoHideWhenHovered && !BrowserUtils.isMobile) {
      // Track hover status of child components
      UIUtils.traverseTree(this, (component) => {
        // Do not track hover status of child containers or spacers, only of 'real' controls
        if (component instanceof Container || component instanceof Spacer) {
          return;
        }

        // Subscribe hover event and keep a count of the number of hovered children
        component.onHoverChanged.subscribe((_, args) => {
          if (args.hovered) {
            hoverStackCount++;
          } else {
            hoverStackCount--;
          }
        });
      });
    }
```

Reply:
Updated as suggested.